sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/jdbc/JdbcUtils.scala

```diff
@@ -456,8 +456,17 @@ object JdbcUtils extends Logging {
 
         case StringType =>
           (array: Object) =>
-            array.asInstanceOf[Array[java.lang.String]]
-              .map(UTF8String.fromString)
+            array match {
+              case _: Array[java.lang.String] =>
+                array.asInstanceOf[Array[java.lang.String]]
+                  .map(UTF8String.fromString)
+              case _: Array[java.util.UUID] =>
+                array.asInstanceOf[Array[java.util.UUID]]
+                  .map(uuid => UTF8String.fromString(uuid.toString))
+              case _ =>
```

Comment:
I can't currently take a close look now (it's mobile) but is it safe (or related) to allow the conversion from obj to string?

Reply:
Will it be more appropriate to use `obj.asInstanceOf[String]` to convert?

Reply:
I was wrong, org.postgresql.util.PGobject cannot be cast to java.lang.String.